src/cpp/lpnamer/LinkProfileReader.cpp

```diff
@@ -29,4 +29,24 @@ LinkProfile LinkProfileReader::ReadLinkProfile(const std::string& filename){
     infile.close();
 
     return result;
-}
\ No newline at end of file
+}
+
+const std::map<std::string, LinkProfile> LinkProfileReader::getLinkProfileMap(const std::string& capacity_folder, const std::vector<CandidateData>& candidateList)
+{
```

Comment:
Should pass a `std::vector<std::string>` instead of a `std::vector<CandidateData>`